services/eventsapi/main.py

```diff
@@ -1,18 +1,25 @@
 from typing import Literal, Union
```

Comment:
Can you try targeting `kafka:29092` instead of `localhost:9092` to see if that works? The reason being that from the `eventsapi` docker container `localhost` is the container itself (versus from your desktop the broker is exposed on `localhost` via a port mapping), but you can reach the broker via `kafka` as DNS will get resolved. In addition, the port number for "internal" traffic 29092 vs 9092 for external traffic.